### ISSUE TITLE
Optimal params after autotune

### ISSUE THREAD

**Issue by phongvis** [facebookresearch/fastText#913]
After running `model = fasttext.train_supervised(input='cooking.train', autotuneValidationFile='cooking.valid')`, does the `model` object contain information about the optimal parameters? As in the documentation, the model is retrained with optimal params. However, when I inspect parameters through `model`, I only see default values such as for `lr`, `epoch`, `minCount`, etc.

Another related question: does the model retrain with a merge of train and validation sets or just a train set?

Many thanks.

**Comment by phongvis** [comment]
Thank you for your prompt reply. I look forward to that new feature.

**Comment by ChiragSoni95** [comment]
Yes, it will be nice to view the optimal characters  selected after auto-tuning. Looking forward to it!

**Comment by ChiragSoni95** [comment]
Yes. I agree. It will be nicer if we can view the optimal hyper parameters after auto-tuning on validation file. Look forward to it!

**Comment by Allenlaobai7** [comment]
#887

**Comment by sherbold** [comment]
Autotune looks like a really nice and convenient feature. Thanks for that. 

However, no access to the best params restricts the use cases and is also not nice for research papers, because it is not possible to report the model parameters. Moreover, I cannot use a cross-validation approach, because I cannot train with the best params on all data. I could only determine the best CV model and could use that. 

Would it be possible to at least expose the best params as part of the text output, while no good solution to communicate or store them otherwise is available?

**Comment by arimbr** [comment]
Hi @Celebio. I found a way to get the model parameters with the Python bindings by inspecting the model object with `model.f.getArgs()`.

I use the following code in this [repository](https://github.com/arimbr/valohai-fasttext-example/blob/master/models/classification/commands.py) to retrain the model on all the data.

```
train_parameters = {
    'lr': 0.1,
    'dim': 100,
    'ws': 5,
    'epoch': 5,
    'minCount': 1,
    'minCountLabel': 0,
    'minn': 0,
    'maxn': 0,
    'neg': 5,
    'wordNgrams': 1,
    'bucket': 2000000,
    'thread': multiprocessing.cpu_count() - 1,
    'lrUpdateRate': 100,
    't': 1e-4,
    'label': LABEL_SEPARATOR,
    'verbose': 2,
    'pretrainedVectors': '',
    'seed': 0,
}

def get_model_parameters(model):
    args_getter = model.f.getArgs()

    parameters = {}
    for param in train_parameters:
        attr = getattr(args_getter, param)
        if param == 'loss':
            attr = attr.name
        parameters[param] = attr

    return parameters
```
